inc/StyleUtils.php

```diff
@@ -100,6 +92,152 @@ public function cssStyleini($tpl, $preview=false) {
+            '__highlight__' => "#ff9",
+            '__link__' => "#00f",
+        );
+
+        $combined = [];
+
+        if (!is_array($config_cascade[$type])) {
+            trigger_error('Missing config cascade for "'.$type.'"',E_USER_WARNING);
+        }
+
+        foreach (array('default','local','protected') as $config_group) {
+
+            if (empty($config_cascade[$type][$config_group])) continue;
+
+            foreach ($config_cascade[$type][$config_group] as $file) {
+                // replace the placeholder with the name of the current template
+                $file = str_replace('TPL_PLACEHOLDER', $conf['template'], $file);
+
+                if (file_exists($file)) {
+                    $config = call_user_func_array($fn, array_merge(array($file), $params));
+
+                    if (is_array($config['stylesheets'])) {
+                        foreach($config['stylesheets'] as $file => $mode) {
+
+                            // set proper server dirs
+                            $webbase = $this->getServerPath($config_group);
+
+                            // validate and include style files
+                            $stylesheets = array_merge($stylesheets, $this->getValidatedStyles($stylesheets, $file, $mode, $incbase, $webbase));
+
+                            // FIXME do we also need to fix paths as in the replacements section below?
+                            $combined['stylesheets'] = is_array($combined['stylesheets']) ? array_merge($combined['stylesheets'], $stylesheets) : $stylesheets;
+                        }
+                    }
+
+                    if (is_array($config['replacements'])) {
+                        $replacements = array_replace($replacements, $this->cssFixreplacementurls($config['replacements'], $webbase));
+                        $combined['replacements'] = array_merge($replacements, $config['replacements']);
+
+                    }
+                }
+            }
+        }
+
+        return $combined;
+    }
+
+
+    protected function getValidatedStyles($stylesheets, $file, $mode, $incbase, $webbase) {
+        global $conf;
+        if (!file_exists($incbase . $file)) {
+            list($extension, $basename) = array_map('strrev', explode('.', strrev($file), 2));
+            $newExtension = $extension === 'css' ? 'less' : 'css';
+            if (file_exists($incbase . $basename . '.' . $newExtension)) {
+                $stylesheets[$mode][$incbase . $basename . '.' . $newExtension] = $webbase;
+                if ($conf['allowdebug']) {
+                    msg("Stylesheet $file not found, using $basename.$newExtension instead. Please contact developer of \"{$conf['template']}\" template.", 2);
+                }
+            }
+        }
+        $stylesheets[$mode][$incbase . $file] = $webbase;
+        return $stylesheets;
+    }
+
+    protected function getServerPath($config_group) {
+        if ($config_group === 'default') {
+            return tpl_basedir();
```

Comment:
I guess `$tpl` should be passed here as well. I guess it would be a good idea to make the template a class member passed in the constructor.

Reply:
The PR now includes a constructor with `$tpl` and `$preview` as class members. All Dokuwiki code that uses the class has also been adjusted in commit 4845eff54